code/game/machinery/seed_extractor.dm

```diff
@@ -8,6 +8,43 @@
 
 /obj/structure/machinery/seed_extractor/attackby(obj/item/O as obj, mob/user as mob)
 
+	// Plant bag and other storage containers.
+	if(istype(O,/obj/item/storage))
+		var/obj/item/storage/container = O
+		if(container.contents.len > 0)
+			to_chat(user, SPAN_NOTICE("You start dumping the contents of [container] into [src]."))
+			if(!do_after(user, 1.5 SECONDS, INTERRUPT_ALL, BUSY_ICON_GENERIC))
+				return
+			for(var/obj/item/item in container)
```

Comment:
since you're typechecking below, you can add "as anything" to scik the typecheck for loop does.

so essentially this 
for(var/obj/item/item as anything in container)

I could be wrong but it seems very logical no?